### PR TITLE
Add utp transfer to find content return

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
Add support for boolean `utpTransfer` field in json responses to find content & recursive find content jsonrpc requests.

### How was it fixed?
Repurpose `ContentInfo` response type to accommodate both find content and recursive find content responses. As well as adding the `utp_transfer` field.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] support inside `trin-beacon`
- [x] add new field to spec (https://github.com/ethereum/portal-network-specs/pull/221)
- [x] Add entry to the [release notes](https://github.com/ethereum/trin/blob/master/newsfragments/README.md) (may forgo for trivial changes)
- [x] Clean up commit history

### CODE REVIEW

**Comment by njgheorghita** [comment]
https://github.com/ethereum/portal-network-specs/pull/221 is merged so this is no longer blocked and ready for :eyes:

File: trin-history/src/jsonrpc.rs [hunk 1]
```diff
@@ -124,11 +123,15 @@ async fn recursive_find_content(
 
     // If tracing is not required, return content.
     if !is_trace {
-        return Ok(content_response_string);
+        return Ok(json!(ContentInfo::Content {
+            content: serde_json::from_value(content_response_string).map_err(|e| e.to_string())?,
```
Comment (carver) [inline trin-history/src/jsonrpc.rs:127]: Something here seems really off. Are we encoding to json string in `content_response_string`, then decoding it to re-encode it? ... I'm seeing now that maybe the same thing is happening in the trace below, but that also seems wrong.

Does this work?
```rust
    // Handle empty content
    let content_string = match possible_content_bytes {
        Some(bytes) => hex_encode(bytes),
        None => CONTENT_ABSENT.to_string(), // "0x"
    };

    // If tracing is not required, return content.
    if !is_trace {
        return Ok(json!(ContentInfo::Content {
            content: content_string,
```

File: trin-validation/src/oracle.rs [hunk 2]
```diff
@@ -51,18 +52,37 @@ impl HeaderOracle {
...
-        let hwp_ssz = hwp_ssz
-            .as_str()
-            .ok_or_else(|| anyhow!("Invalid HWP format."))?;
-        let hwp_ssz = hex_decode(hwp_ssz)?;
-        HeaderWithProof::from_ssz_bytes(&hwp_ssz)
-            .map_err(|err| anyhow!("Invalid HWP received from chain history network: {err:?}"))
+        let hwp = match serde_json::from_value(hwp)? {
+            ContentInfo::Content { content, .. } => content,
+            ContentInfo::ConnectionId { .. } => {
+                return Err(anyhow!(
```
Comment (carver) [inline trin-validation/src/oracle.rs:64]: I think you might be able to do roughly the same pattern as my comment before with "Nice cleanup".

**Review by carver — APPROVED** [review]
No blockers!

**Pull request merged**